### PR TITLE
Bump php to 8.0

### PR DESCRIPTION
**Pull request by VincentLanglet**
## Changelog

```markdown
### Removed
- Support for PHP 7.4
```

### CODE REVIEW

**Comment by VincentLanglet** [comment]
Need https://github.com/sonata-project/dev-kit/pull/2044

File: composer.json [hunk 1]
```diff
@@ -42,13 +42,13 @@
         "psalm/plugin-phpunit": "^0.16",
         "psalm/plugin-symfony": "^3.0",
         "rector/rector": "^0.12",
-        "symfony/config": "^4.4 || ^5.3 || ^6.0",
-        "symfony/dependency-injection": "^4.4 || ^5.3 || ^6.0",
-        "symfony/http-foundation": "^4.4 || ^5.3 || ^6.0",
-        "symfony/http-kernel": "^4.4 || ^5.3 || ^6.0",
+        "symfony/config": "^5.4 || ^6.0",
```
Comment (jordisala1991) [inline composer.json:45]: would be nice to also drop dbal 2

File: src/Handler.php [hunk 2]
```diff
@@ -17,14 +17,8 @@
 
 final class Handler
 {
-    private \Iterator $source;
-
-    private WriterInterface $writer;
-
-    public function __construct(\Iterator $source, WriterInterface $writer)
+    public function __construct(private \Iterator $source, private WriterInterface $writer)
```
Comment (jordisala1991) [inline src/Handler.php]: multiline

File: src/Source/AbstractPropertySourceIterator.php [hunk 3]
```diff
@@ -34,42 +34,22 @@ abstract class AbstractPropertySourceIterator implements \Iterator
...
-     * @var string[]
-     */
-    protected $fields = [];
+    protected PropertyAccessor $propertyAccessor;
 
     /**
      * @param string[] $fields Fields to export
      */
-    public function __construct(array $fields, string $dateTimeFormat = 'r')
+    public function __construct(protected array $fields, protected string $dateTimeFormat = 'r')
```
Comment (jordisala1991) [inline src/Source/AbstractPropertySourceIterator.php]: multiline

File: src/Source/AbstractPropertySourceIterator.php [hunk 4]
```diff
@@ -108,11 +84,11 @@ public function getDateTimeFormat(): string
     }
 
     /**
-     * @param object|mixed[] $current
+     * @phpstan-param object|array<mixed> $current
```
Comment (jordisala1991) [inline src/Source/AbstractPropertySourceIterator.php]: imo it can remain as param

File: src/Source/AbstractXmlSourceIterator.php [hunk 5]
```diff
@@ -22,113 +22,70 @@
...
+    protected bool $currentRowEnded = false;
 
-    /**
-     * @var int
-     */
-    protected $position = 0;
+    protected int $position = 0;
 
-    public function __construct(string $filename, bool $hasHeaders = true)
+    public function __construct(protected string $filename, protected bool $hasHeaders = true)
```
Comment (jordisala1991) [inline src/Source/AbstractXmlSourceIterator.php]: multiline

File: src/Source/CsvSourceIterator.php [hunk 6]
```diff
@@ -49,38 +39,23 @@ final class CsvSourceIterator implements \Iterator
...
-        bool $hasHeaders = true
-    ) {
-        $this->filename = $filename;
-        $this->delimiter = $delimiter;
-        $this->enclosure = $enclosure;
-        $this->escape = $escape;
-        $this->hasHeaders = $hasHeaders;
+    private array | false $currentLine = [];
+
+    public function __construct(private string $filename, private string $delimiter = ',', private string $enclosure = '"', private string $escape = '\\', private bool $hasHeaders = true)
```
Comment (jordisala1991) [inline src/Source/CsvSourceIterator.php]: multiline

File: src/Source/DoctrineDBALConnectionSourceIterator.php [hunk 7]
```diff
@@ -21,42 +21,31 @@
...
+     *
      * @var Result
      */
     private $result;
 
     /**
      * @param mixed[] $parameters
      */
-    public function __construct(Connection $connection, string $query, array $parameters = [])
+    public function __construct(private Connection $connection, private string $query, private array $parameters = [])
```
Comment (jordisala1991) [inline src/Source/DoctrineDBALConnectionSourceIterator.php]: multiline

File: src/Source/DoctrineODMQuerySourceIterator.php [hunk 8]
```diff
@@ -19,24 +19,20 @@ final class DoctrineODMQuerySourceIterator extends AbstractPropertySourceIterato
 {
     private Query $query;
 
-    private int $batchSize;
-
     /**
      * @param array<string> $fields Fields to export
      */
-    public function __construct(Query $query, array $fields, string $dateTimeFormat = \DateTimeInterface::ATOM, int $batchSize = 100)
+    public function __construct(Query $query, array $fields, string $dateTimeFormat = \DateTimeInterface::ATOM, private int $batchSize = 100)
```
Comment (jordisala1991) [inline src/Source/DoctrineODMQuerySourceIterator.php]: multiline

File: src/Source/DoctrineORMQuerySourceIterator.php [hunk 9]
```diff
@@ -15,35 +15,28 @@
...
 {
     protected Query $query;
 
-    private int $batchSize;
-
     /**
      * @param array<string> $fields Fields to export
      */
-    public function __construct(Query $query, array $fields, string $dateTimeFormat = 'r', int $batchSize = 100)
+    public function __construct(Query $query, array $fields, string $dateTimeFormat = 'r', private int $batchSize = 100)
```
Comment (jordisala1991) [inline src/Source/DoctrineORMQuerySourceIterator.php]: multiline

File: src/Source/SymfonySitemapSourceIterator.php [hunk 10]
```diff
@@ -21,36 +21,17 @@
...
-        \Iterator $source,
-        RouterInterface $router,
-        string $routeName,
-        array $parameters = []
-    ) {
-        $this->source = $source;
-        $this->router = $router;
-        $this->routeName = $routeName;
-        $this->parameters = $parameters;
+    public function __construct(private \Iterator $source, private RouterInterface $router, private string $routeName, private array $parameters = [])
```
Comment (jordisala1991) [inline src/Source/SymfonySitemapSourceIterator.php]: multiline

File: src/Source/XmlSourceIterator.php [hunk 11]
```diff
@@ -20,18 +20,12 @@
  */
 final class XmlSourceIterator extends AbstractXmlSourceIterator
 {
-    private string $mainTag;
-
-    private string $dataTag;
-
-    public function __construct(string $filename, string $mainTag = 'datas', string $dataTag = 'data')
+    public function __construct(string $filename, private string $mainTag = 'datas', private string $dataTag = 'data')
```
Comment (jordisala1991) [inline src/Source/XmlSourceIterator.php]: multiline

File: src/Writer/FormattedBoolWriter.php [hunk 12]
```diff
@@ -20,17 +20,8 @@
...
 final class FormattedBoolWriter implements WriterInterface
 {
-    private WriterInterface $writer;
-
-    private string $trueLabel;
-
-    private string $falseLabel;
-
-    public function __construct(WriterInterface $writer, string $trueLabel = 'yes', string $falseLabel = 'no')
+    public function __construct(private WriterInterface $writer, private string $trueLabel = 'yes', private string $falseLabel = 'no')
```
Comment (jordisala1991) [inline src/Writer/FormattedBoolWriter.php]: multiline

File: src/Writer/GsaFeedWriter.php [hunk 13]
```diff
@@ -47,12 +39,8 @@ final class GsaFeedWriter implements WriterInterface
      * @param string       $datasource A datasouce
      * @param string       $feedtype   A feedtype (full|incremental|metadata-and-url)
      */
-    public function __construct(\SplFileInfo $folder, string $dtd, string $datasource, string $feedtype)
+    public function __construct(private \SplFileInfo $folder, private string $dtd, private string $datasource, private string $feedtype)
```
Comment (jordisala1991) [inline src/Writer/GsaFeedWriter.php]: multiline

File: src/Source/AbstractPropertySourceIterator.php (hunk 4 above)
Comment (VincentLanglet) [reply src/Source/AbstractPropertySourceIterator.php]: Currently not https://github.com/rectorphp/rector/issues/7186

Comment (jordisala1991) [reply src/Source/AbstractPropertySourceIterator.php]: Can you add a todo or something, it might be worth changing it once rector releases the fix

**Comment by VincentLanglet** [comment]
Done @jordisala1991

File: src/Source/AbstractPropertySourceIterator.php (hunk 3 above)
Comment (jordisala1991) [reply src/Source/AbstractPropertySourceIterator.php]: This one is missing

File: src/Writer/XmlExcelWriter.php [hunk 14]
```diff
@@ -50,12 +41,11 @@ final class XmlExcelWriter implements WriterInterface
...
-    public function __construct(string $filename, bool $showHeaders = true, $columnsType = null)
-    {
-        $this->filename = $filename;
-        $this->showHeaders = $showHeaders;
-        $this->columnsType = $columnsType;
-
+    public function __construct(
+        private string $filename,
+        private bool $showHeaders = true,
+        private $columnsType = null
```
Comment (jordisala1991) [inline src/Writer/XmlExcelWriter.php]: isnt this mixed?

File: src/Source/DoctrineDBALConnectionSourceIterator.php [hunk 15]
```diff
@@ -21,42 +21,34 @@
...
     /**
      * @var array<string, mixed>|false
      */
-    private $current;
+    private array | false $current = false;
 
     private int $position = 0;
 
     /**
+     * Add type hint when dropping support for Doctrine DBAL < 3.1.
```
Comment (jordisala1991) [inline src/Source/DoctrineDBALConnectionSourceIterator.php]: We might do ^3.1 then?

File: src/Source/AbstractXmlSourceIterator.php (hunk 5 above)
Comment (jordisala1991) [reply src/Source/AbstractXmlSourceIterator.php]: This one is missing

File: src/Source/DoctrineODMQuerySourceIterator.php (hunk 8 above)
Comment (jordisala1991) [reply src/Source/DoctrineODMQuerySourceIterator.php]: and this

File: src/Source/DoctrineORMQuerySourceIterator.php (hunk 9 above)
Comment (jordisala1991) [reply src/Source/DoctrineORMQuerySourceIterator.php]: and this

File: src/Source/AbstractPropertySourceIterator.php (hunk 3 above)
Comment (VincentLanglet) [reply src/Source/AbstractPropertySourceIterator.php]: Indeed, it should be better now

File: src/Source/XmlSourceIterator.php (hunk 11 above)
Comment (jordisala1991) [reply src/Source/XmlSourceIterator.php]: and this

**Review by jordisala1991 — DISMISSED** [review]

**Pull request merged**